src/util/setSelection.ts

```diff
@@ -5,20 +5,23 @@
   @param offset    Character offset of selection.
   @param end       If true, sets the offset to the end of the text.
  */
-export const setSelection = (el, { offset, end } = {}) => {
+type SelectionOptionsType = {offset: number, end: number}
+export const setSelection = (el: HTMLElement, { offset, end }: SelectionOptionsType = {offset: 0, end: 0}) => {
   if (el.childNodes.length === 0) {
     el.appendChild(document.createTextNode(''))
   }
   const textNode = el.firstChild
   const range = document.createRange()
-  const sel = window.getSelection()
+  const sel = window.getSelection() || new Selection()
   // automatically constrain offset to text length
   // this may still throw an error if the text node does no exist any longer
-  try {
-    range.setStart(textNode, offset ? Math.min(offset, textNode.textContent.length) : end ? textNode.textContent.length : 0)
-  }
-  catch (e) {
-    console.warn(e)
+  if(textNode!==null) {
```

Comment:
I believe the `textNode` is guaranteed to be valid since it is appended at the beginning of the function.

Reply:
That's right, but the compiler doesn't quite understand that. 
Maybe we can refactor our code to make it more explicit. Something like this helps to get rid of the compiler error - 
```
const textNode = el.firstChild || el.appendChild(document.createTextNode(''))
```